### PR TITLE
Allow Initial Sync to Work with Simulator 

### PR DESCRIPTION
**Pull request by nisdas**
This would allow initial sync to function with a simulator. By allowing the simulator to respond to requests by block slot number instead of block hash it will allow a node to be able to sync up to the latest chain head. This should also help to close out #661.

- [x] Modify simulator so that it responds to requests by slot number.
- [x] Modify the p2p message sent by the simulator, so that it can be used by either a node performing normal sync or initial sync
- [x] Add tests

### CODE REVIEW

File: beacon-chain/db/state.go [hunk 1]
```diff
@@ -102,6 +102,22 @@ func (db *BeaconDB) GetCrystallizedState() (*types.CrystallizedState, error) {
...
+// SaveCrystallizedState updates the crystallized state for initial sync.
+func (db *BeaconDB) SaveCrystallizedState(cState *types.CrystallizedState) error {
+
+	return db.update(func(tx *bolt.Tx) error {
+		chainInfo := tx.Bucket(chainInfoBucket)
+		cStateEnc, err := cState.Marshal()
+		if err != nil {
+			return err
+		}
+		if err := chainInfo.Put(cStateLookupKey, cStateEnc); err != nil {
```
Comment (rauljordan) [inline beacon-chain/db/state.go]: return chainInfo.Put(...)

**Comment by nisdas** [comment]
Hey guys made the requested changes, now any p2p message more than 100 characters is truncated. And simulator now works fine, I just had to tweak the `lastStateRecalc` in the simulator's crystallized state so that the condition does not fail. I went up to 100 blocks and a new node synced with it immediately lol

**Comment by rauljordan** [comment]
Good stuff - excited for the next PR to be able to continue from the same data dir each time

**Comment by terenc3t** [comment]
@nisdas sorry there are a few more issues.

Looking at the run log from your last commit, BlockChain service never ran. After slot 5, it's only p2p service and simulator.
Log: https://gist.github.com/terenc3t/bc3f6dd89b321a09d9aa1af1eefb5e57

Here is the normal log:
https://gist.github.com/terenc3t/0374442ec7b48ab9f5100590fbad7f7c

**Comment by prestonvanloon** [comment]
Blocked by #787

**Comment by prestonvanloon** [comment]
Lint issues:
```

beacon-chain/simulator/service.go:271:26: error strings should not be capitalized or end with punctuation or a newline
--
  | beacon-chain/simulator/service.go:276:26: error strings should not be capitalized or end with punctuation or a newline
  | beacon-chain/simulator/service.go:281:26: error strings should not be capitalized or end with punctuation or a newline
  | beacon-chain/simulator/service.go:286:26: error strings should not be capitalized or end with punctuation or a newline


```

File: beacon-chain/sync/initial-sync/service.go [hunk 2]
```diff
@@ -44,8 +46,9 @@ type Config struct {
 // CrystallizedStateBufferSize determines the buffer size of thhe `crystallizedStateBuf` channel.
 func DefaultConfig() Config {
 	return Config{
-		SyncPollingInterval:         1 * time.Second,
+		SyncPollingInterval:         time.Duration(int64(params.BeaconConfig().SlotDuration)) * time.Second * 4,
```
Comment (terenc3t) [inline beacon-chain/sync/initial-sync/service.go]: What is this constant 4? Can we predefined this constant?

Comment (rauljordan) [reply beacon-chain/sync/initial-sync/service.go]: Fixed into a config var

**Pull request merged**